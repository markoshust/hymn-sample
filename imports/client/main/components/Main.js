import React, { Component } from 'react';
import Hello from './Hello';
import Info from './Info';

class Main extends Component {
  state = {
    counter: 0,
  };

  handleClick = () => {
    this.setState({ counter: this.state.counter + 1 });
  };

  render() {
    return (
      <div>
        <h1>Welcome to Meteor!</h1>
        <Hello
          counter={this.state.counter}
          onClick={this.handleClick}
        />
        <Info />
      </div>
    );
  }
}

export default Main;
