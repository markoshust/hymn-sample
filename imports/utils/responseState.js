const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
