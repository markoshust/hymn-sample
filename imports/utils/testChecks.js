import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
