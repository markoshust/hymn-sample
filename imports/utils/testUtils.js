import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
