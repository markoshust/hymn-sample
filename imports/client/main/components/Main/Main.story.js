import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Main from './Main';

const stories = storiesOf('Main', module);

stories.add('Main', () => (
  <Main />
));
