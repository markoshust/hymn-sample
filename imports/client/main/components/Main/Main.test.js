import React from 'react';
import { renderCheck } from '/imports/utils/testUtils';
import Main from './Main';

describe('<Main />', () => {
  it('renders correctly', () => {
    renderCheck(<Main />);
  });
});
