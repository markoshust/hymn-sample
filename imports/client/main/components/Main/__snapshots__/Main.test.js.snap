exports[`<Main /> renders correctly 1`] = `
<div>
  <h1>
    Welcome to Meteor!
  </h1>
  <div>
    <button
      onClick={[Function onClick]}>
      Click Me
    </button>
    <p>
      You've pressed the button 
      0
       times.
    </p>
  </div>
  <div>
    <h2>
      Learn Meteor!
    </h2>
    <ul>
      <li>
        <a
          href="https://www.meteor.com/try"
          rel="noopener noreferrer"
          target="_blank">
          Do the Tutorial
        </a>
      </li>
      <li>
        <a
          href="http://guide.meteor.com"
          rel="noopener noreferrer"
          target="_blank">
          Follow the Guide
        </a>
      </li>
      <li>
        <a
          href="https://docs.meteor.com"
          rel="noopener noreferrer"
          target="_blank">
          Read the Docs
        </a>
      </li>
      <li>
        <a
          href="https://forums.meteor.com"
          rel="noopener noreferrer"
          target="_blank">
          Discussions
        </a>
      </li>
    </ul>
  </div>
</div>
`;
